src/material/icon/icon.ts

```diff
@@ -250,6 +250,11 @@ export class MatIcon extends _MatIconMixinBase implements OnInit, AfterViewCheck
     // Update font classes because ngOnChanges won't be called if none of the inputs are present,
     // e.g. <mat-icon>arrow</mat-icon> In this case we need to add a CSS class for the default font.
     this._updateFontIconClasses();
+
+    // Add appropriate MDC class if it is inside a MDC button to be styled properly
+    if (this._elementRef.nativeElement.parentElement?.classList.contains('mdc-button')) {
+      this._elementRef.nativeElement.classList.add('mdc-button__icon');
```

Comment:
In general, we try to avoid dependencies between components, especially between MDC and non-MDC versions. I see a couple of alternatives:
1. Redeclare the styles for `.mdc-button__icon`, but have it target `.mat-mdc-button .mat-icon` instead. I'm guessing that MDC has a mixin for it already.
2. Have the MDC-based button look for icons instead. This is a bit fragile, because the icon can come in at a later point.

Reply:
I think this PR had fixed it at one point: https://github.com/angular/components/pull/20963 did something change that broke it again?